packages/frontend/core/src/modules/app-tabs-header/views/app-tabs-header.tsx

```diff
@@ -197,31 +325,40 @@ export const AppTabsHeader = ({
         {pinned.map(tab => {
           return (
             <WorkbenchTab
+              dnd={mode === 'app'}
               tabsLength={pinned.length}
               key={tab.id}
               workbench={tab}
+              onDrop={data => onDrop(data, tab.id)}
               active={tab.active}
             />
           );
         })}
         {pinned.length > 0 && unpinned.length > 0 && (
           <div className={styles.pinSeparator} />
         )}
-        {unpinned.map(workbench => {
+        {unpinned.map(tab => {
           return (
             <WorkbenchTab
+              dnd={mode === 'app'}
               tabsLength={tabs.length}
-              key={workbench.id}
-              workbench={workbench}
-              active={workbench.active}
+              key={tab.id}
+              workbench={tab}
+              onDrop={data => onDrop(data, tab.id)}
+              active={tab.active}
             />
           );
         })}
+      </div>
+      <div
+        className={styles.spacer}
+        ref={spacerDropTargetRef}
+        data-dragging-over={draggedOver}
```

Comment:
![CleanShot 2024-08-01 at 18.33.49@2x.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/g3jz87HxbjOJpXV3FPT7/4e53c155-a8be-420d-911d-7c5014ab504d.png)

`data-dragged-over` is automatic attribute